.gitignore

```diff
@@ -1,28 +1,50 @@
-node_modules/
+#####################################################################################
+# Generated
+#####################################################################################
 
-# Ignore build directories
-build/
-dist/
+## Packages installed by NPM. https://docs.npmjs.com/about-packages-and-modules
+node_modules/
 
-# Ignore cdk related directories
-.cdk.staging
-cdk.out
+## Vitest coverage output. https://vitest.dev/guide/coverage.html
+coverage/
 
-# Ignore Turbo directory
-.turbo
+## Turborepo output. https://turbo.build/repo/docs/getting-started/existing-monorepo#edit-gitignore
+.turbo/
 
-# Ignore .env files
-*.env
+# Logs.
+*.log
+.pnpm-debug.log*
 
-# Ignore various editor-specific files
-.idea
-.vscode
 
-# Ignore various log files
-npm-debug.log*
-yarn-debug.log*
-yarn-error.log*
-*.log.*
+#####################################################################################
+# Operating System
+#####################################################################################
 
-# Misc
+## MacOS folder settings. https://en.wikipedia.org/wiki/.DS_Store
 .DS_Store
+
+## Windows folder icon settings. https://learn.microsoft.com/en-us/windows/win32/shell/how-to-customize-folders-with-desktop-ini#use-a-desktopini-file
+Desktop.ini
+
+
+#####################################################################################
+# Development Environment
+#####################################################################################
+
+## Private environment variables.
+.env
+.env.*
+!.env.example
+
+## JetBrains IDE settings. https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+.idea/*.iml
+.idea/workspace.xml
+.idea/usage.statistics.xml
+.idea/shelf
+
+## VSCode settings. Based on the following recommendation: https://stackoverflow.com/a/57749909
+.vscode/**/*
+!.vscode/settings.json
+!.vscode/tasks.json
+!.vscode/launch.json
+!.vscode/extensions.json
```

Comment:
I feel like it's worth just ignoring all of these do people actually share settings like this through git

Reply:
Also what was this generated by just curious

Reply:
Stop reading the gitignore! :nerd_face: I wrote it all from scratch myself :sunglasses: